src/coldfront_plugin_api/scim_v2/groups.py

```diff
@@ -117,3 +125,26 @@ def _set_user_status_on_allocation(allocation, user, status):
                 status=AllocationUserStatusChoice.objects.get(name=status)
             )
         return au
+    
+    @staticmethod
+    def _set_user_on_project(project, user, status, role, enable_notifications):
+        pu = ProjectUser.objects.filter(
+            project=project,
+            user=user
+        ).first()
+
+        if pu:
+            pu.status = ProjectUserStatusChoice.objects.get(name=status)
+            pu.role = ProjectUserRoleChoice.objects.get(name=role)
+            pu.enable_notifications = enable_notifications
```

Comment:
This line would override the `enable_notifications` value that the ProjectUser already has. Any particular reason why you're doing this?

Reply:
@knikolla I didn't realize that. Actually, this makes me want to ask...
Currently I've written the function such that any users who are assigned to allocation are added to the project **with the assumption that they're a user (not a PI, etc) and don't want to be notified**. And as of how I've written it, these will be the assumptions whether the user was never assigned to the project (the if clause evaluates false), or was (if clause is true). 

Do you think that it's fine for the false block to make these assumptions? I'll change the true block to only change status.

Reply:
The assumption of the `User` role is correct. As for notifications, I believe it should be set to true.